apps/designer/app/canvas/hovered-instance-connector.ts

```diff
@@ -0,0 +1,101 @@
+import { useEffect } from "react";
+import debounce from "lodash.debounce";
+import { idAttribute } from "@webstudio-is/react-sdk";
+import { utils } from "@webstudio-is/project";
+import { publish, subscribe } from "~/shared/pubsub";
+import { useRootInstance } from "~/shared/nano-states";
+import { subscribeScrollState } from "~/shared/dom-hooks";
+
+export const useHoveredInstanceConnector = () => {
+  const [rootInstance] = useRootInstance();
+
+  useEffect(() => {
+    if (rootInstance === undefined) {
+      return;
+    }
+
+    let hoveredElement: undefined | Element = undefined;
+
+    const publishHover = debounce((element: Element) => {
+      const id = element.getAttribute(idAttribute) ?? undefined;
+      if (id === undefined) {
+        return;
+      }
+      const instance = utils.tree.findInstanceById(rootInstance, id);
+      if (instance === undefined) {
+        return;
+      }
+      // @todo combine hover events
+      publish({
+        type: "hoveredInstanceRect",
+        payload: element.getBoundingClientRect(),
+      });
+      publish({
+        type: "hoverInstance",
+        payload: {
+          id: instance.id,
+          component: instance.component,
+        },
+      });
+    }, 50);
```

Comment:
why debounce 50 and not 150 or 200? Is it needed at all for scroll start/end?

Reply:
It's not for scroll. Allows to avoid glitchy delay on mouseover and skip fast hovers